### ISSUE TITLE
preprocess.py extraDataset

### ISSUE THREAD

**Issue by celinew1221** [stanfordnlp/mac-network#7]
First of all, I am trying to use extraDataset. There is one bug I found in preprocessData function:
`for tier in extraData` should be `extraDataset`

Secondly, it is unclear how we can use extra options, especially what extraVal means. Does it mean that only extra validation set is used in validation, but the code seems to only train on the validation set? So does it mean that the model only train on the validation set of the extra data?

**Comment by dorarad** [comment]
Hi, thanks a lot for pointing to the bug, will fix!
I'm working on submission to a conference in the next couple of days so will get back to this right afterwards!

**Comment by dorarad** [comment]
alright fixed the bug. The original goal of the extra flags it to finetune/test on clevr-human datasets.
For any of the extra options, you first have to pass the --extra and --extraVal flags:
--extra: needed to enable any of the extra options.
--exteaVal: needed to use the same images for both the standard (CLEVR) and extra (CLEVR-humans) datasets - for sure this isn't the best flag name, I'll fix that!
--trainExtra: needed if you'd like to finetune on the extra dataset, otherwise it will be used only for testing.
--finetuneNum: should be used to specify the number of instances you'd like to fine tune on (only in case trainExtra is used)
--alterExtra: an alternative to trainExtra is alterExtra - if you use that, then instead of finetuning the model will be trained together on the standard and extra data, when they are alternated in between: e.g. X batches of standard data and then 1 batch of extra data and the repeating. This option tends to give better results for CLEVR-humans (just training from scratch a model using this option).
--alterNum: used to set the ratio between standard and extra data, each alterNum (X) batches of standard data, 1 extra data batch will be inserted. 

Hope this helps! Let me know if you have any further questions!
